central/search/service/service_impl.go

```diff
@@ -195,84 +194,94 @@ func RunAutoComplete(ctx context.Context, queryString string, categories []v1.Se
 	}
 	// Set the max return size for the query
 	query.Pagination = &v1.QueryPagination{
-		Limit: maxAutocompleteResults,
+		Limit: pageSize,
 	}
 
 	if len(categories) == 0 {
 		categories = autocompleteCategories.AsSlice()
 	}
-	var autocompleteResults []autocompleteResult
-	for _, category := range categories {
-		if category == v1.SearchCategory_ALERTS && !shouldProcessAlerts(query) {
-			continue
-		}
-		searcher, ok := searchers[category]
-		if searcher == nil {
-			if ok {
-				utils.Should(errors.Errorf("searchers map has an entry for category %v, but the returned searcher was nil", category))
-			}
-			return nil, errors.Wrapf(errox.InvalidArgs, "Search category %q is not implemented", category.String())
-		}
 
-		optMultiMap := categoryToOptionsMultimap[category]
-		if optMultiMap == nil {
-			return nil, errors.Wrapf(errox.InvalidArgs, "Search category %q is not implemented", category.String())
+	resultSet := set.NewStringSet()
+	for i := 0; i < maxRetries; i++ {
```

Comment:
How I understand this change is that we fetch the auto-complete result (max 10), remove all duplicates and may be left with fewer results. Wouldn't it be easier to just fetch more results and return max 10 by reducing the result set to 10 in Go code?

nit: Could you document why the retry is necessary?

Reply:
grab 100 and then reduce to 10.  makes some sense.  

Looks like there may be some mechanisms to get distinct on arrays and the jsonb objects that we use to store the maps.  Might be worth looking at as that would solve it entirely

Reply:
got it.  Just grabbing 100 or so seems to be the simplest path then.  That is a small enough number that it should not cause any issues.

Would it be worth writing a spike to ROX-24279?  Probably wouldn't be a huge priority, but worth documenting yet another limitation to how our framework builds queries.

Reply:
Changed it to just grab 100 results. And created [ROX-26368](https://issues.redhat.com/browse/ROX-26368)